src/Microsoft.AspNet.Identity/TaskExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Identity
+{
+    public static class TaskExtensions
+    {
+        public static async Task<IdentityResult> WithLoggingAsync<TUser>(this Task<IdentityResult> identityResult, UserManager<TUser> userManager,TUser user,
+            [System.Runtime.CompilerServices.CallerMemberName] string methodName = "") where TUser :class
+        {
+            return await (await identityResult).WithLoggingAsync(userManager, user, methodName);
+        }
+
+        public static async Task<IdentityResult> WithLoggingAsync<TRole>(this Task<IdentityResult> identityResultDelegate, RoleManager<TRole> roleManager, TRole role,
```

Comment:
Because we were awaiting on it, I thought it was good to put the delegate in the name. Now I feel delegate makes sense only if it is a `Func` or an `Action`. I'll change it